public/app/features/explore/PrometheusListView/RawListContainer.tsx

```diff
@@ -0,0 +1,163 @@
+    listRef.current?.resetAfterIndex(0, true);
+  }, [isExpandedView]);
+
+  const calculateInitialHeight = (length: number): number => {
+    const maxListHeight = 600;
+    const shortListLength = 10;
+
+    if (length < shortListLength) {
+      let sum = 0;
+      for (let i = 0; i < length; i++) {
+        sum += getListItemHeight(i, true);
+      }
+
+      return Math.min(maxListHeight, sum);
+    }
+
+    return maxListHeight;
+  };
+
+  const getListItemHeight = (itemIndex: number, isExpandedView: boolean) => {
+    const singleLineHeight = 32;
+    const additionalLineHeight = 22;
+    if (!isExpandedView) {
+      return singleLineHeight;
+    }
+    const item = items[itemIndex];
+
+    // Height of 1.5 lines, plus the number of non-value attributes times the height of additional lines
+    return 1.5 * singleLineHeight + (Object.keys(item).length - valueLabels.length) * additionalLineHeight;
+  };
+
+  return (
+    <section>
+      <header className={styles.header}>
+        <Button
+          variant="secondary"
+          role={'button'}
+          title={`${isExpandedView ? 'Contract results' : 'Expand results'}`}
+          icon={isExpandedView ? 'minus' : 'plus'}
+          type="button"
+          onClick={onContentClick}
+        >
+          {isExpandedView ? 'Contract' : 'Expand'}
+        </Button>
+        <div>Result series: {items.length}</div>
+      </header>
+
+      <div role={'table'}>
+        {
+          <>
+            {/* Show the value headings above all the values, but only if we're in the contracted view */}
+            {valueLabels.length > 1 && !isExpandedView && (
+              <ItemLabels valueLabels={valueLabels} expanded={isExpandedView} />
+            )}
+            <List
+              ref={(list) => {
+                listRef.current = list;
+              }}
+              itemCount={items.length}
+              className={isExpandedView ? styles.mobileWrapper : styles.wrapper}
+              itemSize={(index) => getListItemHeight(index, isExpandedView)}
+              height={calculateInitialHeight(items.length)}
+              width="100%"
+            >
+              {({ index, style }) => {
+                let filteredValueLabels: Field[] | undefined;
+                if (isExpandedView) {
+                  filteredValueLabels = valueLabels.filter((valueLabel) => {
+                    const itemWithValue = items[index][valueLabel.name];
+                    return itemWithValue && itemWithValue !== RawPrometheusListItemEmptyValue;
+                  });
+                }
+
+                return (
+                  <div role="row" style={{ ...style, overflow: 'hidden' }}>
+                    <RawListItem
+                      isExpandedView={isExpandedView}
+                      valueLabels={filteredValueLabels}
+                      totalNumberOfValues={valueLabels.length}
+                      listKey={index}
```

Comment:
Again here I think index is not a reliable key

Reply:
Fixed!